src/main/java/com/sublinks/sublinksapi/authorization/services/RolePermissionService.java

```diff
@@ -203,14 +232,64 @@ public <X extends Throwable> void isPermitted(final Person person,
       final RolePermissionInterface rolePermission, Supplier<? extends X> exceptionSupplier)
       throws X {
 
+    if (person != null && person.isDeleted()) {
+      throw exceptionSupplier.get();
+    }
+
     final Role role = person == null ? roleService.getDefaultGuestRole(
         () -> new RuntimeException("No Guest role found.")) : person.getRole();
 
+    isPermitted(role, rolePermission, exceptionSupplier);
+  }
+
+  /**
+   * Checks if the given person is permitted to perform the specified role permission.
+   *
+   * @param person         The person to check. Can be null.
+   * @param rolePermission The permission to check.
+   * @param communityId    The ID of the community to check.
+   * @return True if the person is permitted, false otherwise.
+   */
+  public boolean isPermitted(final Person person, final RolePermissionInterface rolePermission,
+      final Long communityId) {
+
+    if (person != null && person.isDeleted()) {
+      return false;
+    }
+    final Role role = person == null ? roleService.getDefaultGuestRole(
+        () -> new RuntimeException("No Guest role found.")) : isBannedInCommunity(person,
+        communityId) ? this.roleService.getBannedRole()
+        .orElseThrow(() -> new RuntimeException("No Banned role found.")) : person.getRole();
```

Comment:
ternary inside a ternary is fairly hard to read. Could you break this up instead?